documentation/phased-ranking.html

```diff
@@ -0,0 +1,127 @@
+---
+# Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+title: "Phased Ranking "
+---
+
+<p>
+Rank scores in general become more accurate by using complex expressions
+which use many features, complex features or large tensors.
+Even though Vespa is optimized for such calculations,
+complex expressions become expensive when calculated for each selected document.
+For this reason Vespa can be configured to run multiple phases of ranking expressions -
+This provides a better CPU budget distribution
+by dedicating more resources to the best candidates as calculated per ranking phase.</p>
+<p> To get an overview of how queries are executed in Vespa reading the <a href="querying-vespa.html">querying Vespa</a> guide is recommended
+before diving deep into this document.</p>
+ 
+<p> Vespa supports multiple rank phases:
+
+<ul>
+<li>Re-ranking of the top hits from the content nodes using a custom <a href="searcher-development.html">searcher component</a>. 
+</li>
+<li>Re-ranking at each content node using two phase ranking configured by rank profiles with ranking expressions working on scalars, tensors or matching features.
+</li>
+<li>Ranking with query operators like <a href="https://dl.acm.org/citation.cfm?id=956944">WAND/Weak And</a> which speeds up retrieval of top matches of a disjunction query (OR) </li>
+</ul>
+
+<h2 id="containe-reranking">Container top matches re-ranking</h2>
```

Comment:
containe-reranking -> container-reranking

Reply:
Should we list regular two-phase ranking first in this document? By listing container ranking first I am afraid users might start using that instead of regular two-phase ranking. Perhaps you also should say something about when we recommend using this compared to regular two-phase ranking.